integrations/weights_bias/src/haystack_integrations/tracing/weave/tracer.py

```diff
@@ -0,0 +1,222 @@
+import contextlib
+import os
+from collections.abc import Iterator
+from typing import Any, Optional, Union
+
+from haystack import logging
+from haystack.tracing import Span, Tracer
+from haystack.tracing import tracer as haystack_tracer
+from haystack.tracing.utils import coerce_tag_value
+
+import weave
+from weave.trace.weave_client import Call, WeaveClient
+
+logger = logging.getLogger(__name__)
+
+
+class WeaveSpan(Span):
+    """
+    A bridge between Haystack's Span interface and Weave's Call object.
+
+    Stores metadata about a component execution and its inputs and outputs, and manages the attributes/tags
+    that describe the operation.
+    """
+
+    def __init__(
+        self, call: Optional[Call] = None, parent: Optional[Call] = None, operation: Optional[str] = None
+    ) -> None:
+        self._call = call
+        self._parent = parent
+        self._operation = operation
+        self._attributes: dict[str, Any] = {}
+
+    def set_tag(self, key: str, value: Any) -> None:
+        """
+        Set a tag by adding it to the call's inputs.
+
+        :param key: The tag key.
+        :param value: The tag value.
+        """
+        self._attributes[key] = value
+
+    def set_tags(self, tags: dict[str, Any]) -> None:
+        for k, v in tags.items():
+            self.set_tag(k, v)
+
+    def raw_span(self) -> Any:
+        """Access to the underlying Weave Call object."""
+        return self._call
+
+    def get_correlation_data_for_logs(self) -> dict[str, Any]:
+        """Correlation data for logging."""
+        if not self._call:
+            return {}
+        return {"weave.call_id": self._call.id, "weave.run_id": getattr(self._call, "run_id", "")}
+
+    def set_call(self, call: "weave.Call") -> None:
+        self._call = call
+
+    def get_attributes(self) -> dict[str, Any]:
+        return self._attributes
+
+
+class WeaveTracer(Tracer):
+    """
+    Implements a Haystack's Tracer to make an interface with Weights and Bias Weave.
+
+    It's responsible for creating and managing Weave calls, and for converting Haystack spans
+    to Weave spans. It creates spans for each Haystack component run.
+    """
+
+    def __init__(self, project_name: str) -> None:
+        """
+        Initialize the WeaveTracer.
+
+        :param project_name: The name of the project to trace, this is will be the name appearing in Weave project.
+        """
+
+        content_tracing_enabled = os.getenv("HAYSTACK_CONTENT_TRACING_ENABLED", "false").lower()
+        haystack_tracer.is_content_tracing_enabled = content_tracing_enabled == "true"
```

Comment:
@davidsbatista does this solve the import order problem that we talked about yesterday? If so, we should keep that in mind for the langfuse connector.

Reply:
not 100& sure, but I think regardless of the import order if this variable is not set traces will not be generated

Reply:
by variable, I mean ªhaystack_tracer.is_content_tracing_enabled`

Reply:
I did a test and I think something is odd, here, run a script to send traces and both versions work, regardless of the env variable being set or not.

```bash
WANDB_API_KEY="..." python weights_bias/example/hybrid_retrieval.py
```

```bash
HAYSTACK_CONTENT_TRACING_ENABLED=true WANDB_API_KEY="..." python weights_bias/example/hybrid_retrieval.py
```

Reply:
So you are sending content in both cases? Because basic traces are sent either way, just the content (input/output of components) should be missing.

Reply:
Yes, I see the content, in Weaver, for both cases, by content I mean for instance, the question/string and the retrieved documents.